compiler/rustc_hir_analysis/src/variance/constraints.rs

```diff
@@ -249,6 +249,20 @@ impl<'a, 'tcx> ConstraintContext<'a, 'tcx> {
                 self.add_constraints_from_ty(current, typ, variance);
             }
 
+            ty::Pat(typ, pat) => {
+                match *pat {
+                    ty::PatternKind::Range { start, end, include_end: _ } => {
+                        if let Some(start) = start {
+                            self.add_constraints_from_const(current, start, variance);
```

Comment:
I think pattern kinds should be invariant in their consts, I think you need to do `self.invariant(variance)`

Reply:
Please add some variances_of tests to make sure that this is the case

Reply:
Actually nvm, I guess it doesn't matter 🤔